dtls/src/crypto/mod.rs

```diff
@@ -300,7 +300,12 @@ fn verify_signature(
         SignatureAlgorithm::Rsa if hash_algorithm.hash == HashAlgorithm::Sha1 => {
             &ring::signature::RSA_PKCS1_1024_8192_SHA1_FOR_LEGACY_USE_ONLY
         }
-        SignatureAlgorithm::Rsa if hash_algorithm.hash == HashAlgorithm::Sha256 => {
+        SignatureAlgorithm::Rsa if (hash_algorithm.hash == HashAlgorithm::Sha256)
+            && (remote_key_signature.len() < 256) => {
```

Comment:
Shouldn't this be an exact match for 128 bytes rather than `< 256`.

Reply:
My question was more if `RSA_PKCS1_1024_8192_SHA256_FOR_LEGACY_USE_ONLY` will work with a key length that isn't exactly `1024`. Reading the [docs](https://docs.rs/ring/0.16.20/ring/signature/static.RSA_PKCS1_1024_8192_SHA256_FOR_LEGACY_USE_ONLY.html) it does say that the key length is variable(between 1024 and 8129 bits) so I guess it will. Sounds like we should keep the current conditions.